packages/devextreme/js/__internal/ui/calendar/m_calendar.ts

```diff
@@ -891,23 +956,24 @@ class Calendar<
     }
   }
 
-  _injectComponent(func) {
-    const that = this;
-    return function (params) {
-      extend(params, { component: that });
-      return func(params);
+  _injectComponent<T extends { date: Date; view: string }>(
+    func: (params: T & { component: Calendar<TProperties> }) => boolean,
+  ): (params: T) => boolean {
+    return (params: T): boolean => {
+      const paramsWithComponent = { ...params, component: this };
+      return func(paramsWithComponent as T & { component: Calendar<TProperties> });
```

Comment:
idk, it looks sus to me that we have such a code for a goal "add one more prop to param" - wouldn't it look better if we just carry it inline?

Reply:
hm, yea ty, managed to simplify it https://github.com/DevExpress/DevExtreme/pull/30599/commits/f8f2bf24138a6ed27b92dceaf4ad4b556f536c90